### PR TITLE
change use of errors.Wrap to fmt.Errorf with %w verb

### LINKED ISSUE #4603: change use of errors.Wrap to fmt.Errorf with %w verb
## Summary

Currently, throughout the repo we use errors.Wrap. This change was made before `%w` verb was introduced in go. 

## Proposal

Change all occurrences of `errors.Wrap` to `fmt.Errorf` with the `%w` verb.

____

#### For Admin Use

- [x] Not duplicate issue
- [x] Appropriate labels applied
- ~~[ ] Appropriate contributors tagged~~
- [ ] Contributor assigned/self-assigned

### PR DESCRIPTION
**Pull request by melekes**
Closes #4603

Commands used (VIM):

```
:args `rg -l errors.Wrap`
:argdo normal @q | update
```

where q is a macros rewriting the `errors.Wrap` to `fmt.Errorf`.

### CODE REVIEW

**Comment by erikgrinaker** [comment]
I'm still seeing some use of `github.com/pkg/errors`, i.e. `errors.Errorf()` (which should be `fmt.Errorf()`), and `errors.New` (which can just use the std error package instead). Maybe we should fix it all while we're at it?

File: privval/utils.go [hunk 1]
```diff
@@ -15,10 +14,11 @@ import (
 // report that a connection timeout occurred. This detects both fundamental
 // network timeouts, as well as ErrConnTimeout errors.
 func IsConnTimeout(err error) bool {
-	switch errors.Cause(err).(type) {
-	case EndpointTimeoutError:
+	_, ok := errors.Unwrap(err).(timeoutError)
```
Comment (marbar3778) [inline privval/utils.go:17]:
```suggestion
	 ok := errors.Is(err, timeoutError)
```
?

**Comment by marbar3778** [comment]
you could also enable goerr113 linter in `.golangci.yml` this may cause more changes as the rules are:

```

    - error values should not be compared directly but with errors.Is() method.

    - error should not be created dynamically from scratch but by the wrapping the static (package-level) error.
```

File: privval/utils.go (hunk 1 above)
Comment (melekes) [reply privval/utils.go:17]: nope. Is requires pointer to a struct. timeoutError is an interface

**Comment by melekes** [comment]
Modified everything except `lite` package, which will be removed shortly after.

**Comment by melekes** [comment]
> you could also enable goerr113 linter in .golangci.yml this may cause more changes as the rules are:

let's do in a separate PR maybe

**Pull request merged**